### PR TITLE
Add Polynomial multiplication using FFT

### PR DESCRIPTION
**Pull request by likecs**
Added implementation of Polynomial Multiplication using Fast Fourier Transforms. The implementation in C++ currently uses templates so that it can be easily extended. Also, the codes have been tested for efficiency on online judges like codechef, codeforces, spoj etc.

### CODE REVIEW

File: code/mathematical-algorithms/multiply_polynomial/FFT.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		for (int i = 0; i < n; ++i) cc.push_back(aa[i] * bb[i]);
+		DFT(cc, true);
+		vector<T> ans(sc);
+		for (int i = 0; i < sc; ++i) ans[i] = cc[i].get_real() + 0.5;
+		return ans;
+	}
+};
+
+int main() {
+	vector<int> poly_1(4);		//1 + 3x^2 + 2x^3
```
Comment (arnavb) [inline code/mathematical-algorithms/multiply_polynomial/FFT.cpp]: Why not just initialize the vector on one line? It is pretty redundant to assign every element on a separate line.

File: code/mathematical-algorithms/multiply_polynomial/FFT.cpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	poly_1[0] = 1;
+	poly_1[1] = 0;
+	poly_1[2] = 3;
+	poly_1[3] = 2;
+
+	vector<int> poly_2(3);		//2 + x + x^2
```
Comment (arnavb) [inline code/mathematical-algorithms/multiply_polynomial/FFT.cpp]: Same here.

Comment (likecs) [reply code/mathematical-algorithms/multiply_polynomial/FFT.cpp]: HI, using vector<int> poly_1(4) = {1, 0, 3, 2}; will only work for C++11 onwards. It doesn't work for previous versions. Hence, I used the above method.

Comment (likecs) [reply code/mathematical-algorithms/multiply_polynomial/FFT.cpp]: https://stackoverflow.com/a/38739297/5161197

Comment (arnavb) [reply code/mathematical-algorithms/multiply_polynomial/FFT.cpp]: Ok, that works.

Comment (AdiChat) [reply code/mathematical-algorithms/multiply_polynomial/FFT.cpp]: I see. The comment in the code makes it clear 👍

**Comment by AdiChat** [comment]
Looks good 👍 

Hope you are enjoying your journey with **OpenGenus**

**Pull request merged**